### ISSUE TITLE
Error with datetime

### ISSUE THREAD

**Issue by rgabrielson11** [andrewramsay/ical_to_gcal_sync#38]
Hello,

Trying installing the script for the first time today.  Getting this error in the logs:

Error retrieving iCal data (can't compare offset-naive and offset-aware datetimes)

Any thoughts?

**Comment by andrewramsay** [comment]
Based on the exception text it could be something going wrong inside the icalevents package that the script uses to parse iCal sources. 

If you can share a minimal example of the source that is producing the error I'd be happy to take a look at it and see if I can find the root cause. Also if you can tell me which Python version and platform you're running the script on that would be useful to know.

**Comment by rgabrielson11** [comment]
Hi, on python 3.9.2

Ill work on a dataset that I can send over

**Comment by dm9** [comment]
Not sure if we are looking at the same cause, but I get the same error from an .ics where the DTSTART timezone is recognized (hence offset-aware) but the DTEND timezone is defined only in a VTIMEZONE component that appears in the .ics after the reference to it (and therefore icalendar ignores the timezone altogether and renders it offset-naive). Basically, it appears that icalendar.Component.from_ical( ) cannot cope with forward references to VTIMEZONEs. I think the fix to icalendar is not hard, but I can't get its out-of-the-box test cases to run successfully so I haven't gotten very far with that.

**Comment by dm9** [comment]
Yep, I've confirmed that the latest version of icalendar.Calendar.from_ical still chokes when a VEVENT relies on a forward reference to a VTIMEZONE. That should probably be fixed over in icalendar. But meanwhile there's a simple workaround to get ical_to_gcal_sync to cope with this. Namely, in get_current_events, call icalevents.icalevents.events() twice (ignoring away exceptions the first time). The first call will get the VTIMEZONE into the cache and the second time will find the VTIMEZONEs in the cache and produce the requested events without throwing an exception.

I could provide a pull request with this fix, even though it's not pretty. @andrewramsay, any interest?

**Comment by andrewramsay** [comment]
I'm not really using this script any more but if it fixes the problem I'm happy to merge a PR for it, doesn't sound like it should affect anything else.

**Comment by dm9** [comment]
Actually I worked with collective/icalendar and this (at least the problem I was seeing) is now fixed in their v6.1.3. So an update to your requirements.txt saying icalendar==6.1.3 might do the trick. Not sure if that can be changed without also updating the icalevents requirement... there have also been some incompatible changes in newer versions of icalevents (namely it now wants urllib3 rather than httplib2). 

Anyway, since you said you're not using this script anymore, I'm curious, have you found a better way to sync icalendars over to gcal?

**Comment by andrewramsay** [comment]
I tried messing around with the updated versions of `icalevents` and `icalendar` and I think it's working, try out the version in the `test_icalevents` branch (I haven't tested it with a source that requires authentication though).

> Anyway, since you said you're not using this script anymore, I'm curious, have you found a better way to sync icalendars over to gcal?

Nope, I just changed how I keep track of tasks/todos and found I didn't really need it any more.

**Comment by dm9** [comment]
The test_icalevents branch works for me. Thanks!
